src/main/java/com/foundrysoftware/minecraft/plugins/PluginTemplate.java

```diff
@@ -1,37 +1,171 @@
                 return true;
             }
+            if (label.equalsIgnoreCase("hawkeye")) {
+                if (args[0].equalsIgnoreCase(ARROW_TYPE_TNT)) {
+                    if (sender instanceof Player) {
+                        Player me = (Player) sender;
+                        PlayerInventory inventory = me.getInventory();
+                        for (int i = 0; i < inventory.getSize(); i++) {
+                            ItemStack item = inventory.getItem(i);
+                            if (item == null) {
+                                continue;
+                            }
+                            if (isArrow(item) == true) {
+                                item.setType(Material.TIPPED_ARROW);
+                                item.setAmount(item.getAmount());
+
+                                String potion = "CONDUIT_POWER";
+
+                                item.setItemMeta(createPotion(item,potion));
+                            }
+                        }
+
+                    }
+                } else {
+                    if (sender instanceof Player) {
+                        Player me = (Player) sender;
+                        PlayerInventory inventory = me.getInventory();
+                        for (int i = 0; i < inventory.getSize(); i++) {
+                            ItemStack item = inventory.getItem(i);
+                            if (item == null) {
+                                continue;
+                            }
+                            if (isArrow(item) == true) {
+                                String potion = args[0];
+                                item.setType(Material.TIPPED_ARROW);
+                                item.setAmount(item.getAmount());
+
+                                item.setItemMeta(createPotion(item,potion));
+                            }
+
+                        }
+                    }
+                }
+            }
         }
         return false;
     }
+
+
+    @EventHandler
+    public boolean onShoot(ProjectileHitEvent e) {
+        if (e.getEntity() instanceof Arrow) {
+            Arrow arrow = (Arrow) e.getEntity();
+            Entity shooter = (Entity) arrow.getShooter();
+            Location loc=getLocation(e);
+            Bukkit.getServer().broadcastMessage(loc.toString());
+            if (arrow.hasCustomEffect(PotionEffectType.CONDUIT_POWER)) {
+                loc.getWorld().createExplosion(loc, 10, false, false, shooter);
+            }
+
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isArrow(ItemStack item) {
+        if (item.getType() == Material.TIPPED_ARROW || item.getType() == Material.ARROW || item.getType() == Material.SPECTRAL_ARROW) {
+            return true;
+        } else {
+            return false;
+        }
+
+    }
+
+    private boolean areaDamage(Location loc, int area, int damage, Entity shooter) {
+        int Yarea = area;
```

Comment:
Issue #5 ALL variable and methods names should be camelCase with the first letter being lowercase, its a Java standard. Only class names should be CamelCase but start with an uppercase letter. Constants (final) should be all uppercase with an underscore bwteeen words